brooklyn-tosca-transformer/src/test/java/org/apache/brooklyn/tosca/a4c/brooklyn/ToscaPlanToSpecTransformerTest.java

```diff
@@ -90,6 +96,58 @@ public void testDslInChatApplication() {
 
     @Test
     @SuppressWarnings("unchecked")
+    public void testFullJcloudsLocationDescription() {
+        String templateUrl = getClasspathUrlForResource("templates/full-location.jclouds.tosca.yaml");
+
+        EntitySpec<? extends Application> app = transformer.createApplicationSpec(
+                new ResourceUtils(mgmt).getResourceAsString(templateUrl));
+
+        assertNotNull(app);
+        assertEquals(app.getChildren().size(), 1);
+        EntitySpec<VanillaSoftwareProcess> vanillaEntity =
+                (EntitySpec<VanillaSoftwareProcess>) Iterables.getOnlyElement(app.getChildren());
+
+        assertEquals(vanillaEntity.getLocations().size(), 1);
+        Location location = Iterables.getOnlyElement(vanillaEntity.getLocations());
+        assertTrue(location instanceof JcloudsLocation);
+        assertEquals(((JcloudsLocation) location).getProvider(), "aws-ec2");
+        assertEquals(((JcloudsLocation) location).getRegion(), "us-west-2");
+        assertEquals(((JcloudsLocation) location).getIdentity(), "user-key-id");
+        assertEquals(((JcloudsLocation) location).getCredential(), "user-key");
+    }
+
+    @Test
+    @SuppressWarnings("unchecked")
+    public void testFullByonLocationDescription() {
+        String templateUrl = getClasspathUrlForResource("templates/full-location.byon.tosca.yaml");
+
+        EntitySpec<? extends Application> app = transformer.createApplicationSpec(
+                new ResourceUtils(mgmt).getResourceAsString(templateUrl));
+
+        assertNotNull(app);
+        assertEquals(app.getChildren().size(), 1);
+        EntitySpec<VanillaSoftwareProcess> vanillaEntity =
+                (EntitySpec<VanillaSoftwareProcess>) Iterables.getOnlyElement(app.getChildren());
+
+        assertEquals(vanillaEntity.getLocations().size(), 1);
+        assertTrue(Iterables.getOnlyElement(vanillaEntity.getLocations())
+                instanceof FixedListMachineProvisioningLocation);
+
+        FixedListMachineProvisioningLocation location =
+                (FixedListMachineProvisioningLocation) Iterables
+                        .getOnlyElement(vanillaEntity.getLocations());
+        Map<String, Object> configByon = location.getLocalConfigBag().getAllConfig();
+        assertEquals(configByon.get("user"), "brooklyn");
+        assertEquals(configByon.get("provider"), "byon");
+        assertEquals(configByon.get("provider"), "byon");
+        assertEquals(((List)configByon.get("machines")).size(), 1);
```

Comment:
Thanks a lot. 
Really, I do not know when I should check a `instanceof` or allow to throw an exception.